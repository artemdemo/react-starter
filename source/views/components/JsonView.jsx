import React from 'react';
import Icon from '../../components/Icon/Icon';
import JsonTest from '../../containers/JsonTest/JsonTest';

const JsonView = () => {
    return (
        <React.Fragment>
            <p>
                <Icon name='gift' />
                &nbsp;
                JSON Test View
            </p>
            <JsonTest />
        </React.Fragment>
    );
};

export default JsonView;
