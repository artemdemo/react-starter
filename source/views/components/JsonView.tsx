import React from 'react';
import Icon from '../../components/Icon/Icon';
import JsonTest from '../../containers/JsonTest/JsonTest';

const JsonView = () => {
    return (
        <>
            <p>
                <Icon name='gift' />
                &nbsp;
                JSON Test View
            </p>
            <JsonTest />
        </>
    );
};

export default JsonView;
