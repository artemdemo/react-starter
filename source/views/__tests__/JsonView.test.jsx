import React from 'react';
import renderer from 'react-test-renderer';
import JsonView from '../components/JsonView';

jest.mock('../../components/Icon/Icon');
jest.mock('../../containers/JsonTest/JsonTest');

describe('<JsonView>', () => {
    it('Empty render', () => {
        const tree = renderer.create(
            <JsonView />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
