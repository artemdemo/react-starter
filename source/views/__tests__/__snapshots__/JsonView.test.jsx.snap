// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JsonView> Empty render 1`] = `
Array [
  <p>
    <div
      data-mock="Icon"
    >
      {
  "name": "gift"
}
    </div>
    Â  JSON Test View
  </p>,
  <div
    data-mock="JsonTest"
  />,
]
`;
